src/testing/jest/jest-apis.ts

```diff
@@ -12,30 +12,45 @@
  * file be added to sparingly.
  */
 
+import type { TransformedSource } from '@jest/transform';
 import type { Config } from '@jest/types';
 import { getVersion } from 'jest';
 
+import { Jest28TransformOptions } from './jest-28/jest-preprocessor';
+
 // TODO(STENCIL-959): Improve this typing by narrowing it
 export type JestPuppeteerEnvironment = any;
 
-type Jest26CacheKeyOptions = { instrument: boolean; rootDir: string };
 type Jest26Config = { instrument: boolean; rootDir: string };
 type Jest27TransformOptions = { config: Jest26Config };
-export type JestPreprocessor = {
+
+type BasePreprocessor = {
+  process(sourceText: string, sourcePath: string, ...args: any[]): string | TransformedSource;
+  getCacheKey(sourceText: string, sourcePath: string, ...args: any[]): string;
+};
```

Comment:
Can we just use this as a the type for `JestPreprocessor` to avoid having to make changes to this file with each version added/removed?

Reply:
Yeah, I'm open to that! https://github.com/ionic-team/stencil/pull/4979/commits/879fca506eb0271fe0038d7b149f72659abf8c03